core/interfaces/rebase.py

```diff
@@ -475,34 +532,45 @@ def run_async(self):
             sublime.status_message("Unable to squash first commit.")
             return
 
-        squash_idx, to_squash, before_squash = self.get_idx_entry_and_prev(short_hash)
-        _, _, two_entries_before_squash = self.get_idx_entry_and_prev(before_squash.short_hash)
+        squash_idx, squash_entry, _ = self.get_idx_entry_and_prev(short_hash)
+
+        if self.commit_is_merge(squash_entry.long_hash):
+            sublime.status_message("Unable to squash merges.")
+            return
+
+        self.squash_idx = squash_idx
+        self.squash_entry = squash_entry
+        reversed_logs = list(reversed(self.interface.entries[0:squash_idx]))
+
+        # GsLogBase is a WindowCommand
+        self.window = self.view.window()
+        self.display_commits(self.render_commits(reversed_logs))
```

Comment:
AFAIK, `git_mixins.history` contains only git related code, all UI related codes go to `core.command`.

@divmain Any idea where to put UI related common codes?